_test/test_sqw_class/test_parse_pixel_indices.m

```diff
@@ -0,0 +1,167 @@
+classdef test_parse_pixel_indices < TestCase
+    % Test the sqw class method parse_pixel_indices
+    properties
+        w1
+        w2
+        irun1
+        idet1
+        ien1
+        irun2
+        idet2
+        ien2
+    end
+
+    methods
+        function obj = test_parse_pixel_indices (name)
+            obj = obj@TestCase(name);
+            
+            % Load two tiny sqw objects from file
+            %   w1:  29 pixels, two runs, runid: 100, 200 
+            %        (runid are 100 x header index)
+            %   w2:  18 pixels, three runs, runid: 1000, 2000, 3000 
+            %        (runid are 1000 x header index)
+            load('test_parse_pixel_indices_data.mat','w1','w2')
+            obj.w1 = w1;
+            obj.w2 = w2;
+            
+            obj.irun1 = w1.pix.run_idx(:) / 100;
+            obj.idet1 = w1.pix.detector_idx(:); 
+            obj.ien1 = w1.pix.energy_idx(:);
+            
+            obj.irun2 = w2.pix.run_idx(:) / 1000;
+            obj.idet2 = w2.pix.detector_idx(:); 
+            obj.ien2 = w2.pix.energy_idx(:);
+        end
+        
+        function test_emptySqwObject_ERROR(~)
+            % Must be at least one sqw object in the array
+            w = repmat(sqw(), 0);   % empty sqw array
+            f = @() parse_pixel_indices(w);
+            assertExceptionThrown(f, 'HORACE:parse_pixel_indices:invalid_argument');
+        end
+        
+        function test_SqwObject_noPixels(~)
+            % Scalar sqw object, but without any pixels: is valid
+            w = sqw();
+            [irun,idet,ien]=parse_pixel_indices(w);
+            assertEqual(irun, zeros(0,1))
+            assertEqual(idet, zeros(0,1))
+            assertEqual(ien, zeros(0,1))
+        end
+        
+        function test_scalarSqwObject(obj)
+            % Scalar sqw object
+            w = obj.w1;
+            [irun,idet,ien]=parse_pixel_indices(w);
+            assertEqual(irun, obj.irun1)
+            assertEqual(idet, obj.idet1)
+            assertEqual(ien, obj.ien1)
+        end
+        
+        function test_scalarSqwObject_ipixEmpty_ERROR(obj)
```

Comment:
This test is missing from the cell_array_input tests, because the only tests in cell_array_input concern the correct handling of cell array to object array conversion (including edge cases of empty cell array, for example), and the passing of ipix as a varargin{:} in the internals of the parse_pixel_indices for cell arrays, which just calls the sqw method of the same name. Once the correct passing of the arguments has been tested, all the other tests about size of ipix in relation to the sqw object array in test_parse_pixel_indices apply. The explanation for two files was given earlier.